packages/protocol/contracts/common/FeeCurrencyWhitelist.sol

```diff
@@ -20,10 +20,17 @@ contract FeeCurrencyWhitelist is
 {
   // Array of all the tokens enabled
   address[] public whitelist;
+  // it is not enforce that underlyingTokens in the same order as their underlying
+  // whitelisted address
+  address[] public underlyingTokens;
```

Comment:
We could make this AddressSet which would allow to add underlyingTokens directly in setAdaptor rather than have special method for it

Reply:
This will not work if we want to have multiple adapters for one token ..

Reply:
I'd rather not update this to 0.8 and leave as is

Reply:
The problem with current approach is that it is rather manual. 

You can setAdapter and you can forget to setUnderlyingToken which will result in inconsistent behaviour. If we will introduce AddressSet instead of array, then we can remove `setUnderlyingToken` and instead set the underlying token in `setAdapter`

Reply:
I can make setAdapter also set the underlying token, so it's one call and there's no chance for error.

Reply:
And regarding this will not work for more than a token, yes, that's correct. But in reality I don't see how we could have more than one adaptor for a token,